unit-tests/py/rspy/devices.py

```diff
@@ -603,7 +603,13 @@ def hw_reset( serial_numbers, timeout = 5 ):
         dev = get( sn ).handle
         dev.hardware_reset()
     #
-    _wait_until_removed( serial_numbers )
+    usb_serial_numbers = { device.serial_number for device in _device_by_sn.values() if device.port is not None }
+    if usb_serial_numbers:
+        _wait_until_removed( usb_serial_numbers )
```

Comment:
I don't think so.
for D457 is will wait for disconnection and it will not happen.
This is why I used the usb filter

Reply:
Currently `wait_until_all_ports_disable` will nor work with D457 present.
I'm saying we want to change it so that it has this exact logic in it, including the logic for MIPI.

Reply:
The easiest way is changing `enabled()` to return all non removed devices with port != None.
But it has impact on other calls. line in wait_for, not sure we want wait fore to skip D457.
Don't think I want other changes on this PR.

Reply:
Anyway, I just looked again:
You're waiting for ALL devices to be removed, but you've only done the HW reset on specific SNs...

Reply:
You are right, fixed.